src/main/java/usecases/CourseEvaluator/EvaluatorInteractor.java

```diff
@@ -0,0 +1,77 @@
+package usecases.CourseEvaluator;
+
+
+import entities.Course;
+import entities.CourseFactory;
+
+public class EvaluatorInteractor implements EvaluatorInputBoundary {
+   private EvaluatorOutputBoundary presenter;
+   private StudentDataAccess studentDataAccess;
+
+   private CourseDataAccess courseDbGateway;
+
+   private EvaluatorSessionDataAccess sessionDataAccess;
+
+   private CourseFactory courseFactory;
+
+   private CheckAnswer grade;
+
+
+   public EvaluatorInteractor(EvaluatorOutputBoundary presenter, CourseDataAccess courseDbGateway,
+                              EvaluatorSessionDataAccess sessionDataAccess, CourseFactory courseFactory,
+                              StudentDataAccess studentDataAccess, CheckAnswer grade) {
+        this.presenter = presenter;
+        this.courseDbGateway = courseDbGateway;
+        this.sessionDataAccess = sessionDataAccess;
+        this.courseFactory = courseFactory;
+        this.studentDataAccess = studentDataAccess;
+        this.grade = grade;
+    }
+
+    /**
+     * Use case for checking if a course exists, saving and evaluating  a students grade in percentages. If the course
+     * does not exist, an error is thrown, that informs students that the course does not exist.
+     * if course exists in database
+     * the student grade is saved to the Student data access.
+     * Then the presenter prepares success/fail views depending on each case.
+     *
+     * @param requestModel data structure packaging the courseId.
+     * @return String prepare success and fail view.
+     */
+
+
+    public String evaluate(EvaluatorRequestModel requestModel) {
+
+        if (!sessionDataAccess.courseExists(requestModel.getCourseId())) {
+            return presenter.prepareFailView();
+        }
+        // creates response model for student and course and creates a course for each of them as well
+
+        EvaluatorDbResponseModel studentResponseModel = sessionDataAccess.retrieveCourse(requestModel.getCourseId());
+
+        EvaluatorDbResponseModel courseResponseModel = courseDbGateway.findCourse(requestModel.getCourseId());
```

Comment:
These two lines of code are good. I think we should change the parameters, instead of giving the courseID, we should give the gateway the entire request model. The gateway will be responsible for calling getCourseID and obtaining the course id.